chainstate/tx-verifier/src/transaction_verifier/timelock_check.rs

```diff
@@ -89,13 +94,28 @@ where
         TransactionSourceForConnect::Mempool { current_best } => (*current_best).clone(),
     };
 
+    let mut output_check_required: Option<OutputTimelockCheckRequired> = None;
+
     for input in inputs {
         let outpoint = input.outpoint();
         let utxo = utxos_view
             .utxo(outpoint)
             .map_err(|_| utxo::Error::ViewRead)?
             .ok_or(ConnectTransactionError::MissingOutputOrSpent)?;
 
+        output_check_required = match utxo.output() {
+            TxOutput::Transfer(_, _)
+            | TxOutput::LockThenTransfer(_, _, _)
+            | TxOutput::Burn(_)
+            | TxOutput::CreateDelegationId(_, _) => None,
+            TxOutput::CreateStakePool(_, _) | TxOutput::ProduceBlockFromStake(_, _) => {
+                Some(OutputTimelockCheckRequired::DecommissioningMaturity)
+            }
+            TxOutput::DelegateStaking(_, _) => {
+                Some(OutputTimelockCheckRequired::DelegationSpendMaturity)
+            }
+        };
```

Comment:
I don't like these mutations. I think this can be cleaned up in a better way where we don't mutate a value like this. Here's my suggestion:

1. Iter through inputs' outpoints, map, and pull the utxos, and then put them in a `Vec<(Input, Utxo)>`.
2. Run another iter with a map, and calculate `output_check_required`.
3. Do the other checks.